docs/data/toolpad/core/components/dashboard-layout/dashboard-layout.md

```diff
@@ -78,10 +78,20 @@ Nested navigation structures can be placed in the sidebar as items with the form
 
 ### Navigation Actions
 
-Navigation links have an optional `action` prop that can be used to render any content on the right-side of the respective list item, such as badges with numbers, or buttons to toggle a popover menu.
+Navigation links have an optional `action` prop to render any content on the right-side of the respective list item, such as badges with numbers, or buttons to toggle a popover menu.
 
 {{"demo": "DashboardLayoutNavigationActions.js", "height": 400, "iframe": true}}
 
+### Navigation Pattern Matching
+
+Navigation links have an optional `pattern` prop to define a pattern to be matched for the item to be marked as selected.
+Any path supported by the [path-to-regexp](https://www.npmjs.com/package/path-to-regexp) library can be used, such as:
+
+- Standard paths: `/foo`
+- Named parameters: `/:foo/:bar`
+- Unnamed parameters: `/:foo/(.*)`
+- Optional parameters: `/:foo{/:bar}?`
```

Comment:
We could maybe add demo. a "product" page that has links to show the "next product" and the current path. e.g.

```
current path: /products/20
title: Item 20
[next product](/products/21)
```